src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeOptions.cs

```diff
@@ -54,15 +54,16 @@ public class PipeOptions
             {
                 pauseWriterThreshold = DefaultPauseWriterThreshold;
             }
-            else if (pauseWriterThreshold < 0)
-            {
-                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.pauseWriterThreshold);
-            }
 
             if (resumeWriterThreshold == -1)
             {
                 resumeWriterThreshold = DefaultResumeWriterThreshold;
             }
+
+            if (pauseWriterThreshold < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.pauseWriterThreshold);
+            }
```

Comment:
Not sure why this block needs to move unless we're concerned about `DefaultPauseWriterThreshold` being less than 0. Where it was before would still throw an exception if `pauseWriterThreshold` was `< -1` (due to the `-1` check before it).

Reply:
What I think you want is the `else if` below these lines to just be `if` instead.